nova/virt/vmwareapi/cluster_util.py

```diff
@@ -429,3 +430,26 @@ def update_cluster_das_vm_override(session, cluster, vm_ref, operation='add',
     config_spec.dasVmConfigSpec = [das_vm_spec]
 
     reconfigure_cluster(session, cluster, config_spec)
+
+
+def get_cluster_median_cpu_mhz(session, cluster) -> int:
+    """Get the median CPU frequency for all hosts in a cluster.
+
+    We need this for reservation, since flavors specify reservation in cores,
+    but reservation of a VMware VM's CPUs is specified in Hz. We calculate the
+    median to prevent outliers (esp. hosts that might have much _less_ CPUs for
+    some reason) from skewing the result.
+    """
+    hosts = session._call_method(vutil, "get_object_property", cluster,
+                                 "host")
+    if not hosts:
+        raise CpuReservationNoHostsError(cluster=cluster)
```

Comment:
Is `cluster` a full `moref`? If yes, I'd prefer the variable to show this either via `cluster_ref` or `cluster_ref_value` (if it's not a full MoRef). If it's a full MoRef, I'd also prefer `cluster=vutil.get_moref_value(cluster)` here, because a MoRef renders to string with multiple newlines or as JSON - depending on our `oslo.vwmare`.

Reply:
obsoleted